### ISSUE TITLE
support for wan2.1 mini-batch finetune

### ISSUE THREAD

**Issue by yfpeng1234** [modelscope/DiffSynth-Studio#682]
Hi,
When finetune wan2.1 on my own dataset, I noted that the current finetuning script doesn't support batch training. It seems that the code only support 1sample/GPU. Like the flow matching schedule, noise generation, etc.

**Comment by Shuweis** [comment]
I also meet the same questions, why is set to 1 sample/gpu by default in the training code:"dataloader = torch.utils.data.DataLoader(dataset, shuffle=False, collate_fn=lambda x: x[0])"?  Does it according to the largest memory of GPU?

**Comment by Artiprocher** [comment]
@Shuweis @yfpeng1234 

The reasons are diverse:

- In most cases, a single GPU has limited VRAM, making it difficult to achieve a larger batch size.  
- Dynamic resolution cannot be enabled when batch size > 1.  
- Most community models (e.g., VACE) are not adapted for batch size > 1, and supporting this feature would impose a significant engineering burden.  
- The mathematical equivalent of batch size > 1 can be achieved through gradient accumulation.

**Comment by nankepan** [comment]
5555g4rr54rr4e3we4rewe343ee23332wq221q11`
